# Problem Statement 1:-->

A F&B manager wants to determine whether there is any significant difference in the diameter of the cutlet between two units. A randomly selected sample of cutlets was collected from both units and measured? Analyze the data and draw inferences at 5% significance level. Please state the assumptions and tests that you carried out to check validity of the assumptions.

File: Cutlets.csv

# Business Objectives-->

Quality Control: The primary objective of the F&B manager is to assess whether there is any significant difference in the diameter of the cutlets between the two units. 

Cost Control: Understanding whether one unit produces cutlets with more consistent diameters could influence decisions regarding resource allocation, potentially leading to cost savings.

# Constraints-->

Assumptions of Normality

Quality control

Cost Control

In [57]:
#import required libraries
import pandas as pd
import numpy as np
from scipy import stats

In [58]:
#Load the dataset 
df=pd.read_csv("Cutlets.csv")

In [59]:
#explore the dataset
df.head(5)

,Unit A,Unit B
0,6.8090,6.7703
1,6.4376,7.5093
2,6.9157,6.7300
3,7.3012,6.7878
4,7.4488,7.1522


In [60]:
#shape of dataset
df.shape

(51, 2)

In [61]:
#describe the dataset
df.describe()

,Unit A,Unit B
count,35.000000,35.000000
mean,7.019091,6.964297
std,0.288408,0.343401
min,6.437600,6.038000
25%,6.831500,6.753600
50%,6.943800,6.939900
75%,7.280550,7.195000
max,7.516900,7.545900


In [62]:
df.isnull().sum()

Unit A    16
Unit B    16
dtype: int64

In [63]:
df.isnull().sum()

Unit A    16
Unit B    16
dtype: int64

In [64]:
df.columns

Index(['Unit A', 'Unit B'], dtype='object')

Steps for hypothesis testing-->

1. check null and alternative hypothesis
2. choose a significat labels
3. select a appropriate ststisctical test(t-test(n<30),z-test(n>=30),ANOVO test(when comparing more than 2 groups)-->And theses 3 test are used for comparing mean
and for categorical data (chi-test), for relationship bet variables(peason's corelation, regression analysis)
4. calulate the test statistic
5. find the p val and compare with alpha
p<=alpha -->reject H0
p>alpha -->fail to reject H0
6. conclusion
7.report the result


Null Hypothesis (H₀): There is no significant difference in the mean diameter of cutlets between Unit A and Unit B. (i.e., μA = μB)


Alternative Hypothesis (H₁): There is a significant difference in the mean diameter of cutlets between Unit A and Unit B. (i.e., μA ≠ μB)

Significance Level:

α = 0.05 (5% significance level)

In [66]:
# Extract and clean data
unit_a = df['Unit A'].dropna().astype(float)
unit_b = df['Unit B'].dropna().astype(float)

In [67]:
# Check for normality using the Shapiro-Wilk test
stat_a, p_a = stats.shapiro(unit_a)
stat_b, p_b = stats.shapiro(unit_b)

print(f"Shapiro-Wilk test for Unit A: p-value = {p_a}")
print(f"Shapiro-Wilk test for Unit B: p-value = {p_b}")

Shapiro-Wilk test for Unit A: p-value = 0.31998491287231445
Shapiro-Wilk test for Unit B: p-value = 0.5225146412849426


Unit A: p-value = 0.31998

Unit B: p-value = 0.52251

Since both p-values are greater than 0.05, we fail to reject the null hypothesis. This means that the data for both Unit A and Unit B follows a normal distribution.

In [68]:
# variance test for equality using levene test
stat, p_value = stats.levene(unit_a, unit_b)
print(f"Levene's test p-value = {p_value}")

Levene's test p-value = 0.4176162212502553


Null Hypothesis (H₀): The variances of Unit A and Unit B are equal.
    
Alternative Hypothesis (H₁): The variances of Unit A and Unit B are not equal.

p-value = 0.4176 (which is greater than 0.05)

Fail to reject H₀ → We assume equal variances between Unit A and Unit B.

In [69]:
# If variances are equal, perform a Two-sample t-test (equal variance)
t_stat, p_value = stats.ttest_ind(unit_a, unit_b, equal_var=True)

print(f"T-statistic: {t_stat}, p-value: {p_value}")


T-statistic: 0.7228688704678063, p-value: 0.4722394724599501


Interpretation:
Result= p-value=0.4722 ,which is greater than 0.05

H0: There is no significant difference in the mean diameter of cutlets between Unit A and Unit B
    
H1: There is a significant difference in the mean diameter of cutlets between Unit A and Unit B
    
Since the p-value (0.4722) is greater than 0.05, we fail to reject the null hypothesis (H₀)

Conclusion:  There is no significant difference in the mean diameter of cutlets between Unit A and Unit B.